src/main/scala/org/apache/spark/sql/delta/OptimisticTransaction.scala

```diff
@@ -692,6 +692,14 @@ trait OptimisticTransactionImpl extends TransactionalWrite with SQLMetricsReport
     attemptVersion
   }
 
+  /**
+   * Used to filter collections of `FileAction`s with `dataChange=false`
+   * when `DeltaSQLConf.CONCURRENT_COMPACTION_CONFLICT_DETECTION` is enabled.
+   */
+  protected def filterDataChange(f: FileAction): Boolean = {
```

Comment:
nit: Can you change the name to `isDataChange` or alike. This `filter` can be misleading.

Reply:
`isDataChangeOrConflict` could work for me too.